src/TokenManager/CachedToken.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare(strict_types=1);
+
+namespace MyOnlineStore\GuzzleAuthorizationMiddleware\TokenManager;
+
+use MyOnlineStore\GuzzleAuthorizationMiddleware\Token;
+use Psr\Cache\CacheItemPoolInterface;
+
+final class CachedToken implements TokenManagerInterface
+{
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cachePool;
+
+    /**
+     * @var TokenManagerInterface
+     */
+    private $innerTokenManager;
+
+    public function __construct(CacheItemPoolInterface $cachePool, TokenManagerInterface $innerTokenManager)
+    {
+        $this->cachePool = $cachePool;
+        $this->innerTokenManager = $innerTokenManager;
+    }
+
+    public function getToken(): Token
+    {
+        $item = $this->cachePool->getItem(self::class);
```

Comment:
This will store a single token based on this class name? How will that work for tokens for different tenant ids?

Reply:
The `cachePool` instance provided will have to scope for user/store ID. Since that info is not available here, there is no way to add it right here.

Reply:
Ah right :+1:

Reply:
Another possible solution would be to add an interface that provides the ID, which has to be implemented by the user. Like `UriProviderInterface`.

Reply:
> Actually, I could probably just use the URI provided by that implementation as well.

Indeed, but using the tenant specific cache would be OK as well

Reply:
> Indeed, but using the tenant specific cache would be OK as well

Well, an implementation for `UriProviderInterface` is required for the JWT manager, so it will be available anyway. By using it here to scope the cached token, we can guarantee it is saved correctly without depending on the `cachePool` implementation. I actually like that idea better than being dependant on an external implementation.

TL;DR, I will add `UriProviderInterface` as a dependency and use the uri to scope the cached token.